main/src/ca/uwaterloo/flix/language/phase/jvm/JvmName.scala

```diff
@@ -24,17 +24,27 @@ import java.nio.file.{Path, Paths}
   */
 object JvmName {
 
-  case class MethodDescriptor(arguments: List[JvmType], result: JvmType) {
+  // TODO: Would be nice to allow BackendObjType here to avoid conversions
+  case class MethodDescriptor(arguments: List[BackendType], result: VoidableType) {
     /**
       * Returns the type descriptor of this method.
       */
-    override lazy val toString: String = AsmOps.getMethodDescriptor(arguments, result)
+    override lazy val toString: String = {
```

Comment:
Is this for debugging or for actual code generation?

Reply:
Actual code generation. I was split between making a sensibly named function that returns a string (which makes it possible to use the nonsense `toString`) or use the `toString` which is confusing but now you can not call the wrong thing by accident.

Reply:
We should *never* use `toString`. Give it a proper name.

Please add a line to `STYLE.md` saying never to use toString for anything other than debugging :)